src/fastmcp/tools/tool_transform.py

```diff
@@ -798,3 +829,43 @@ def _function_has_kwargs(fn: Callable[..., Any]) -> bool:
         return any(
             p.kind == inspect.Parameter.VAR_KEYWORD for p in sig.parameters.values()
         )
+
+
+class ToolTransformConfig(FastMCPComponent):
+    """Provides a way to transform a tool."""
+
+    name: str | None = Field(default=None, description="The new name for the tool.")
+
+    arguments: dict[str, ArgTransformConfig] = Field(
+        default_factory=dict,
+        description="A dictionary of argument transforms to apply to the tool.",
+    )
+
+    def apply(self, tool: Tool) -> Tool:
+        """Apply the transform to the tool."""
+
+        tool_changes = self.model_dump(exclude_unset=True, exclude={"arguments"})
+
+        return TransformedTool.from_tool(
+            tool=tool,
+            **tool_changes,
+            transform_args={k: v.to_arg_transform() for k, v in self.arguments.items()},
+        )
+
+    @classmethod
+    def apply_to_tools(
```

Comment:
This feels misplaced to me - its a classmethod but doesn't actually use the class at all? It could more simpler and more discoverable as a documented module function I think